docs/site/hasOne-relation.md

```diff
@@ -0,0 +1,259 @@
+the foreign key property on its associated model. The `hasOne` relation is
+defined on a source model `Supplier` in the example below:
+
+{% include code-caption.html content="/src/models/supplier.model.ts" %}
+
+```ts
+import {Account} from './account.model.ts';
+import {Entity, property, hasOne} from '@loopback/repository';
+
+export class Supplier extends Entity {
+  @property({
+    type: 'number',
+    id: true,
+  })
+  id: number;
+
+  @property({
+    type: 'string',
+    required: true,
+  })
+  name: string;
+
+  @hasOne(() => Account)
+  account?: Account;
+
+  constructor(data: Partial<Supplier>) {
+    super(data);
+  }
+}
+```
+
+On the other side of the relation, we'd need to declare a `belongsTo` relation
+since every account has to belong to exactly one supplier:
+
+```ts
+import {Supplier} from './supplier.model.ts';
+import {Entity, property, belongsTo} from '@loopback/repository';
+
+export class Account extends Entity {
+  @property({
+    type: 'number',
+    id: true,
+  })
+  id: number;
+
+  @property({
+    type: 'string',
+  })
+  accountManager: string;
+
+  @belongsTo(() => Supplier)
+  supplierId: number;
+
+  constructor(data: Partial<Account>) {
+    super(data);
+  }
+}
+```
+
+The definition of the `hasOne` relation is inferred by using the `@hasOne`
+decorator. The decorator takes in a function resolving the target model class
+constructor and optionally a has one relation definition object which can e.g.
+contain a custom foreign key to be stored as the relation metadata. The
+decorator logic also designates the relation type and tries to infer the foreign
+key on the target model (`keyTo` in the relation metadata) to a default value
+(source model name appended with `id` in camel case, same as LoopBack 3).
+
+The decorated property name is used as the relation name and stored as part of
+the source model definition's relation metadata.
+
+A usage of the decorator with a custom foreign key name for the above example is
+as follows:
+
+```ts
+// import statements
+class Supplier extends Entity {
+  // constructor, properties, etc.
+  @hasOne(() => Account, {keyTo: 'suppId'})
+  account?: Account;
+}
+```
+
+## Setting up your database for hasOne relation - MySQL
+
+At the moment, LoopBack does not provide the means to enforce referential
+integrity for the `hasOne` relation. It is up to users to set this up at the
+database layer so constraints are not violated. Let's take MySQL as the backing
+database for our application. Given the `supplier` has one `account` scenario
+above, we need to run two SQL statements on the `account` table for the database
+to enforce referential integrity and align with LoopBack's `hasOne` relation.
+
+1. Make `supplierId` property or column a foreign key which references the `id`
+   from Supplier model's `id` property:
+   - `alter table <databaseName>.Account add foreign key (supplierId) REFERENCES <databaseName>.Supplier(id);`
```

Comment:
Please create a fenced block using triple back-ticks instead of using inline code.

Also I believe the usual convention is to use all-upper-case letters for SQL commands, e.g. `ALTER TABLE`.